### DISCUSSION TITLE
Masking Image

### DISCUSSION THREAD

**Discussion by VB-Byte** [ImageMagick/ImageMagick#2886] [Help] [answered]
Hello people,
When I mix the image and the mask, the result is the image with a dark shadow in the background, what I want is only the image with light, this is the code I am using:

convert face1.bmp mask.bmp -alpha off -compose copy-opacity -composite new.png

ImageMagick Version: 6.8.9-3 q16 x86

![Image](https://user-images.githubusercontent.com/75255296/100611740-8a5fb500-32f0-11eb-8832-d28787d8bffc.png)

Is there a way to do this ?, thanks.

**Comment by snibgo** [comment, marked as answer]
The dark shadow comes from your face1.bmp, which has dark pixels outside your central circle. You should first make those pixels the same yellow as the face. You can probably do that by using mask.bmp. In that image, make all non-white pixels black, so you have a white circle with black background. Then use that as a mask to turn pixels in face1.bmp yellow.

**Reply by VB-Byte** [reply]
snibgo
Thanks for answering, but I already got it, the algorithm is as follows:

convert face1.bmp mask.bmp -alpha Off -fx "v==0 ? 0 : u/v - black/v + black/v" mask.bmp -compose Copy_Opacity -composite new.png

This is exactly how I wanted it, thanks

**Comment by fmw42** [comment]
Please always post original images separately, so that others can work with your data to demonstrate.

**Reply by VB-Byte** [reply]
fmw42
Okay, thanks for the tip.
